libs/libc/signal/sig_raise.c

```diff
@@ -45,5 +45,5 @@
 
 int raise(int signo)
 {
-  return tkill(_SCHED_GETTID(), signo);
+  return tgkill(_SCHED_GETPID(), _SCHED_GETTID(), signo);
```

Comment:
can we save _SCHED_GETPID to local variable? getpid is very time consume in protected and kernel build

Reply:
That is one option, another could be to stash tcb->group->tg_pid into TLS as well. Perhaps put it into struct task_info_s?

https://github.com/apache/nuttx/blob/358469e5bbb5e9507777c8c0b563c761213ececd/sched/tls/tls_initinfo.c#L75

Reply:
yes, but it isn't harmful to add a local variable here to avoid call _SCHED_GETPID twice.

Reply:
The calls are to getpid and gettid, not 2x getpid

Reply:
Or change the interface of tgkill, I don't see any use for PID in there. It should be enough to pass any TID and just a boolean of whether the signal is to be delivered to whole group or specifically to the TID.

I don't even understand why the (linux) tkill and tgkill have been added to NuttX, this is completely redundant to sig_kill and nxsig_kill, only that sig_kill would have needed the group/thread information as well.

Reply:
Having said all that, I don't think that this PR is correct either. "raise" should send the signal explicitly only to the calling thread. I need to still tinker with this. 

I am currently having some issues with the master branch, there is some instability after yesterday or the day before. for example sim:citest sometimes just segfaults aready at the start....

Reply:
Ok. that change to raise was just wrong. I reverted it. Found other bugs and fixed those, I had made mistakes in signal pending set handling. Perhaps the patch is now ok, at least passes tests on my PC (although I had to revert #16231 , because with those changes more or less nothing works for me, I filed an issue on that).